apps/user-profile/src/app/pages/settings-page/settings-mailing-lists/settings-mailing-lists.component.ts

```diff
@@ -90,24 +171,46 @@ export class SettingsMailingListsComponent implements OnInit {
   }
 
   getOptOutAttribute(resource: RichResource): void {
+    this.selectedResource = resource.name;
     this.index = this.resources.indexOf(resource);
     this.optOutAttribute = this.optOuts[this.index].attribute;
   }
 
-  setOptOut(): void {
-    this.optOuts[this.index].attribute.value = this.optOutAttribute.value
-      ? null
-      : ('true' as unknown as object);
+  unsubscribe(): void {
+    this.optOuts[this.index].attribute.value = 'true' as unknown as object;
+    this.attributesManagerService
+      .setMemberResourceAttribute(this.optOuts[this.index])
+      .subscribe(() => {
+        this.notificator.showSuccess(
+          'You are now excluded from the mailing list of ' + this.selectedResource + '.'
+        );
+      });
```

Comment:
I think you should use `en.json` and `cs.json` files for customized translation. In `en.json` you can use somethnig like `"UNSUBSCRIBED": "You are now excluded from the mailing list of {{resource}}."` and translate it also for cs in `cs.json`. In ts file you can use this:
```suggestion
        this.notificator.showSuccess(
          this.translate.instant('OPT_OUT_MAILING_LISTS.UNSUBSCRIBED', {
            resource: this.selectedResource,
          }) as string
        );
      });
```

Reply:
The parameter `resource` in the translation is not needed, since the resource name is not translated.
```suggestion
        this.notificator.showSuccess(
           this.translate.instant('OPT_OUT_MAILING_LISTS.UNSUBSCRIBED') as string 
           + ' ' + this.selectedResource
        );
      });
```